### PR TITLE
drivers: adc: tla202x: add support for tla2022 and tla2024 to tla2021 driver

### PR DESCRIPTION
**Pull request by biglben**
This PR extends the existing TLA2021 driver to support TLA2022 and TLA2024. TLA2022 introduces a programmable gain amplifier (PGA), while TLA2024 adds a multiplexer to the TLA2022 for measuring up to 4 channels

### CODE REVIEW

**Comment by biglben** [comment]
Still relevant. Rebased it to newest main

File: drivers/adc/adc_tla2021.c [hunk 1]
```diff
@@ -197,7 +200,9 @@ static void tla2021_perform_read(const struct device *dev)
 	/*
 	 * Wait until sampling is done
 	 */
+	k_usleep(WAKEUP_TIME_US + CONVERSION_TIME_US);
 	do {
+		k_yield();
```
Comment (bjarki-andreasen) [inline drivers/adc/adc_tla2021.c]: I would move the yield down to right before the condition to not have a sleep (yield) followed by another yield:
```
        k_yield();
} while (!(reg & REG_CONFIG_OS_msk));
```

**Review by bjarki-andreasen — APPROVED** [review]
Really nice work, sorry this PR seems to have been missed completely

**Review by nandojve — APPROVED** [review]
I would add a configuration that allows to run on samples/drivers/adc. This helps to test locally and it is a good entry point to others.

**Pull request merged**